### PR TITLE
Blank requests: Avoid sending blank metrics.

### PR DESCRIPTION
**Pull request by ilya-lt**

### CODE REVIEW

File: tests/test__influx_reporter.py [hunk 1]
```diff
@@ -47,8 +47,7 @@ def test_gauge_without_tags(self):
             autocreate_database=False
         )
 
-        with mock.patch.object(influx_reporter, "_try_send",
-                               wraps=influx_reporter._try_send) as send_mock:
+        with mock.patch.object(influx_reporter, "_try_send") as send_mock:
```
Comment (dgilboa) [inline tests/test__influx_reporter.py]: how is that related to the not report on blanks ?

File: tests/test__influx_reporter.py [hunk 2]
```diff
@@ -21,20 +21,20 @@ def setUp(self):
     def tearDown(self):
         super(TestInfluxReporter, self).tearDown()
 
-    def test_report_now(self):
+    def test_not_called_on_blank(self):
```
Comment (dgilboa) [inline tests/test__influx_reporter.py]: what about the test_report_now - you replaced it with not to call on blank but I assume having a test for report now is still something needed ?

File: tests/test__influx_reporter.py [hunk 3]
```diff
@@ -21,20 +21,20 @@ def setUp(self):
...
             influx_reporter.report_now()
-            patch.assert_called()
+            patch.assert_not_called()
 
     def test_create_database(self):
         r1 = InfluxReporter(registry=self.registry, autocreate_database=True)
         with mock.patch("pyformance.reporters.influx.urlopen") as patch:
             r1.report_now()
-            if patch.call_count != 2:
+            if patch.call_count != 1:
```
Comment (dgilboa) [inline tests/test__influx_reporter.py]: I didn't understand why this needed to change from 2 to 1 - please explain (f2f or here if it's enough)

File: tests/test__influx_reporter.py (hunk 2 above)
Comment (ilya-lt) [reply tests/test__influx_reporter.py]: report now is not suppose to send anything if it doesn't have anything to send, that's the whole point. Sending valid data is covered by a lot of other tests.

File: tests/test__influx_reporter.py (hunk 3 above)
Comment (ilya-lt) [reply tests/test__influx_reporter.py]: it was sending an empty request right after creating a database, it was invalid but not considered bad enough to bother with.

File: tests/test__influx_reporter.py (hunk 1 above)
Comment (ilya-lt) [reply tests/test__influx_reporter.py]: It's not, I can't track down why the test broke now, but it's not related to this pr. Probably some version difference between python 3.7 or anything specific to my environment(this project doesn't have a virtualenv).

Anyway I don't see the purpose of that wraps=, the test is not suppose to actually contact a real server, so it shouldn't be there.

**Comment by ilya-lt** [comment]
[Avoid sending empty requests to pyformance.](https://trello.com/c/3VvcuClS/1313-avoid-sending-empty-requests-to-pyformance)
